lib/Doctrine/ORM/Internal/Hydration/ObjectHydrator.php

```diff
@@ -538,8 +558,8 @@ protected function hydrateRowData(array $row, array &$result)
         // Append scalar values to mixed result sets
         if (isset($rowData['scalars'])) {
             if (! isset($resultKey)) {
-                $resultKey = isset($this->resultSetMapping()->indexByMap['scalars'])
-                    ? $row[$this->resultSetMapping()->indexByMap['scalars']]
+                $resultKey = isset($resultSetMapping->indexByMap['scalars'])
+                    ? $row[$resultSetMapping->indexByMap['scalars']]
                     : $this->resultCounter - 1;
```

Comment:
```suggestion
                $resultKey = $resultSetMapping->indexByMap['scalars'] ?? $this->resultCounter - 1;
```

Reply:
hum does not look correct to me as we are accessing `$row`, not only `$resultSetMapping->indexByMap` ? 🤔